### DISCUSSION TITLE
Top Off manual limit

### DISCUSSION THREAD

**Discussion by Dragonatorul** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#978] [Ideas]
I propose to add the option to set an arbitrary top off limit in the settings. This option would have to check to be above the current regular charge limit, and potentially adjust or disable itself temporarily when the regular limit exceeds the set top off limit. If this option is overwritten by the new regular charge limit, then an indicator should be visible somewhere in the UI to highlight this conflict. 

My use case is that I generally have my MacBook Air M1 plugged in, but move it from one plugged-in location to another throughout the day. Say I have it set in Sailing mode with 20% range with a 60% max charge. That's fine for regular use in this scenario, but on the rare occasion when I anticipate I will need to work off the battery for an extended period of time I would love to have the ability to click the top off button and have it charge only to 90% or 95% instead of full 100%. This would presumably reduce the strain on the battery and also reduce the charge time.

**Comment by zorin** [comment]
+1 for the idea.
@MatthiasKerbl & @davidwernhart what do you think guys?

I'm sure this optional feature goes well with the general 20-80% premise (regular calibration being a different topic on its own). Especially true lately, with the latest M-series powered MacBooks providing outstanding battery longevity (up to 20h) already, some may not see the purpose of squeezing the battery to the very maximum 100% charge (but say 95%, being enough).

IMHO a top-up button being split in to two buttons (top up to x%, top up to 100%), would be an awesome feature to have.
